src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static void setBalance(String fruit, int quantity) {
+        fruitStorage.put(fruit, quantity);
+    }
+
+    public static void addFruit(String fruit, int quantity) {
+        fruitStorage.put(fruit, fruitStorage.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public static void removeFruit(String fruit, int quantity) {
+        fruitStorage.put(fruit, fruitStorage.getOrDefault(fruit, 0) - quantity);
+    }
```

Comment:
Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Right now, the `removeFruit` method can potentially set a negative balance for a fruit, which is not acceptable. You need to add a check to ensure that the balance cannot become negative.